src/pytorch_lightning/serve/servable_module_validator.py

```diff
@@ -0,0 +1,169 @@
+
+        self.optimization = optimization
+        self.host = host
+        self.port = port
+        self.server = server
+        self.timeout = timeout
+        self.exit_on_failure = exit_on_failure
+        self.resp: Optional[requests.Response] = None
+
+    @rank_zero_only
+    def on_train_start(self, trainer: "pl.Trainer", servable_module: "pl.LightningModule") -> None:
+        if isinstance(trainer.strategy, _NOT_SUPPORTED_STRATEGIES):
+            raise Exception(
+                f"The current strategy {trainer.strategy} used "
+                "by the trainer isn't supported for sanity serving yet."
+            )
+
+        if not isinstance(servable_module, ServableModule):
+            raise TypeError(f"The provided model should be subclass of {ServableModule.__qualname__}.")
+
+        if not is_overridden("configure_payload", servable_module, ServableModule):
+            raise NotImplementedError("The `configure_payload` method needs to be overridden.")
+        if not is_overridden("configure_serialization", servable_module, ServableModule):
+            raise NotImplementedError("The `configure_serialization` method needs to be overridden.")
+        if not is_overridden("serve_step", servable_module, ServableModule):
+            raise NotImplementedError("The `serve_step` method needs to be overridden.")
+
+        servable_module.trainer = None
+
+        process = Process(target=self._start_server, args=(servable_module, self.host, self.port, self.optimization))
+        process.start()
+
+        servable_module.trainer = trainer
+
+        ready = False
+        t0 = time.time()
+        while not ready:
+            try:
+                resp = requests.get(f"http://{self.host}:{self.port}/ping")
+                ready = resp.status_code == 200
+            except requests.exceptions.ConnectionError:
+                pass
+            if time.time() - t0 > self.timeout:
+                raise Exception(f"The Server didn't start in {self.timeout}")
+            time.sleep(0.1)
+
+        payload = servable_module.configure_payload()
+
+        if "body" not in payload:
+            raise Exception(f'Your provided payload {payload} should have a field named "body".')
+
+        self.resp = requests.post(f"http://{self.host}:{self.port}/serve", json=payload)
+        process.kill()
+
+        if is_overridden("configure_response", servable_module, ServableModule):
+            response = servable_module.configure_response()
+            if self.resp.json() != response:
+                raise Exception(f"The expected response {response} doesn't match the generated one {self.resp.json()}.")
+
+        if self.exit_on_failure and not self.successful:
+            raise SystemExit("The ServableModel API hasn't been properly implemented.")
+
+        if self.successful:
+            print(f"Your model is servable and the received payload was {self.resp.json()}.")
```

Comment:
This should be a debug statement if anything. Also, use a logger instead of basic print

Reply:
Should be info. If you are using the callback, you want to know it was successful.